check-stage1: improper_ctypes in compiletest/raise_fd_limit on OSX

I'm guessing this is a result of #29546, though I'm not entirely sure why this changed; 

```
$ make -j4 check-stage1-rpass TESTNAME=match-arm-statics
cfg: version 1.6.0-dev (9952f6435 2015-11-10)
cfg: build triple x86_64-apple-darwin
cfg: host triples x86_64-apple-darwin
cfg: target triples x86_64-apple-darwin
cfg: enabling debug assertions (CFG_ENABLE_DEBUG_ASSERTIONS)
cfg: enabling debuginfo (CFG_ENABLE_DEBUGINFO)
cfg: host for x86_64-apple-darwin is x86_64
cfg: os for x86_64-apple-darwin is apple-darwin
cfg: good valgrind for x86_64-apple-darwin is
cfg: using CC=ccache clang (CFG_CC)
cfg: using CXX=clang++ (CFG_CXX)
cfg: enabling valgrind run-pass tests (CFG_ENABLE_VALGRIND_RPASS)
cfg: valgrind-rpass command set to "/usr/local/bin/valgrind" --error-exitcode=100 --fair-sched=try --quiet --soname-synonyms=somalloc=NONE --suppressions=/Users/tamird/src/rust/src/etc/x86.supp --suppressions=/Users/tamird/src/rust/src/etc/apple-darwin.supp --tool=helgrind
cfg: disabling doc build (CFG_DISABLE_DOCS)
cfg: including test rules
cfg: lexer tooling not available, skipping lexer test...
rustc: x86_64-apple-darwin/stage0/lib/rustlib/x86_64-apple-darwin/bin/compiletest
rustc: x86_64-apple-darwin/stage0/lib/rustlib/x86_64-apple-darwin/lib/librustc_front
cp: x86_64-apple-darwin/stage1/lib/libsyntax
/Users/tamird/src/rust/src/compiletest/raise_fd_limit.rs:37:58: 37:70 error: found Rust type `usize` in foreign module, while `libc::c_uint` or `libc::c_ulong` should be used, #[deny(improper_ctypes)] on by default
/Users/tamird/src/rust/src/compiletest/raise_fd_limit.rs:37                   oldp: *mut libc::c_void, oldlenp: *mut libc::size_t,
                                                                                                                     ^~~~~~~~~~~~
/Users/tamird/src/rust/src/compiletest/raise_fd_limit.rs:38:52: 38:64 error: found Rust type `usize` in foreign module, while `libc::c_uint` or `libc::c_ulong` should be used, #[deny(improper_ctypes)] on by default
/Users/tamird/src/rust/src/compiletest/raise_fd_limit.rs:38                   newp: *mut libc::c_void, newlen: libc::size_t) -> libc::c_int;
                                                                                                               ^~~~~~~~~~~~
error: aborting due to 2 previous errors
make: *** [x86_64-apple-darwin/stage0/lib/rustlib/x86_64-apple-darwin/bin/compiletest] Error 101
make: *** Waiting for unfinished jobs....
```

I have a patch for this, PR to follow.

--- a/src/compiletest/raise_fd_limit.rs
+++ b/src/compiletest/raise_fd_limit.rs
@@ -23,25 +23,8 @@ pub unsafe fn raise_fd_limit() {
     use std::mem::size_of_val;
     use std::ptr::null_mut;
 
-    type rlim_t = libc::uint64_t;
-
-    #[repr(C)]
-    struct rlimit {
-        rlim_cur: rlim_t,
-        rlim_max: rlim_t
-    }
-    extern {
-        // name probably doesn't need to be mut, but the C function doesn't
-        // specify const
-        fn sysctl(name: *mut libc::c_int, namelen: libc::c_uint,
-                  oldp: *mut libc::c_void, oldlenp: *mut libc::size_t,
-                  newp: *mut libc::c_void, newlen: libc::size_t) -> libc::c_int;
-        fn getrlimit(resource: libc::c_int, rlp: *mut rlimit) -> libc::c_int;
-        fn setrlimit(resource: libc::c_int, rlp: *const rlimit) -> libc::c_int;
-    }
     static CTL_KERN: libc::c_int = 1;
     static KERN_MAXFILESPERPROC: libc::c_int = 29;
-    static RLIMIT_NOFILE: libc::c_int = 8;
 
     // The strategy here is to fetch the current resource limits, read the
     // kern.maxfilesperproc sysctl value, and bump the soft resource limit for
@@ -51,25 +34,25 @@ pub unsafe fn raise_fd_limit() {
     let mut mib: [libc::c_int; 2] = [CTL_KERN, KERN_MAXFILESPERPROC];
     let mut maxfiles: libc::c_int = 0;
     let mut size: libc::size_t = size_of_val(&maxfiles) as libc::size_t;
-    if sysctl(&mut mib[0], 2, &mut maxfiles as *mut _ as *mut _, &mut size,
+    if libc::sysctl(&mut mib[0], 2, &mut maxfiles as *mut _ as *mut _, &mut size,
               null_mut(), 0) != 0 {
         let err = io::Error::last_os_error();
         panic!("raise_fd_limit: error calling sysctl: {}", err);
     }
 
     // Fetch the current resource limits
-    let mut rlim = rlimit{rlim_cur: 0, rlim_max: 0};
-    if getrlimit(RLIMIT_NOFILE, &mut rlim) != 0 {
+    let mut rlim = libc::rlimit{rlim_cur: 0, rlim_max: 0};
+    if libc::getrlimit(libc::RLIMIT_NOFILE, &mut rlim) != 0 {
         let err = io::Error::last_os_error();
         panic!("raise_fd_limit: error calling getrlimit: {}", err);
     }
 
     // Bump the soft limit to the smaller of kern.maxfilesperproc and the hard
     // limit
-    rlim.rlim_cur = cmp::min(maxfiles as rlim_t, rlim.rlim_max);
+    rlim.rlim_cur = cmp::min(maxfiles as libc::rlim_t, rlim.rlim_max);
 
     // Set our newly-increased resource limit
-    if setrlimit(RLIMIT_NOFILE, &rlim) != 0 {
+    if libc::setrlimit(libc::RLIMIT_NOFILE, &rlim) != 0 {
         let err = io::Error::last_os_error();
         panic!("raise_fd_limit: error calling setrlimit: {}", err);
     }

